vumi/components/message_store.py

```diff
@@ -1000,18 +1071,44 @@ def has_next_page(self):
         return self._index_page.has_next_page()
 
     def __iter__(self):
-        return (self._format_result(r) for r in self._index_page)
-
-    def _format_result(self, result):
-        value, key = result
-        prefix = self._batch_id + "$"
-        if not value.startswith(prefix):
-            raise ValueError(
-                "Index value %r does not begin with expected prefix %r." % (
-                    value, prefix))
-        suffix = value[len(prefix):]
-        timestamp, delimiter, address = suffix.partition("$")
-        if delimiter != "$":
-            raise ValueError(
-                "Index value %r does not match expected format." % (value,))
-        return (key, timestamp, address)
+        return (self._formatter(self._batch_id, r) for r in self._index_page)
+
+
+def key_with_timestamp_formatter(batch_id, result):
+    value, key = result
+    prefix = batch_id + "$"
+    if not value.startswith(prefix):
+        raise ValueError(
+            "Index value %r does not begin with expected prefix %r." % (
+                value, prefix))
+    return (key, value[len(prefix):])
+
+
+def key_with_ts_and_value_formatter(batch_id, result):
+    value, key = result
+    prefix = batch_id + "$"
+    if not value.startswith(prefix):
+        raise ValueError(
+            "Index value %r does not begin with expected prefix %r." % (
+                value, prefix))
+    suffix = value[len(prefix):]
+    timestamp, delimiter, address = suffix.partition("$")
+    if delimiter != "$":
+        raise ValueError(
+            "Index value %r does not match expected format." % (value,))
+    return (key, timestamp, address)
+
+
+def key_with_rts_and_value_formatter(batch_id, result):
+    value, key = result
+    prefix = batch_id + "$"
+    if not value.startswith(prefix):
+        raise ValueError(
+            "Index value %r does not begin with expected prefix %r." % (
+                value, prefix))
+    suffix = value[len(prefix):]
+    timestamp, delimiter, address = suffix.partition("$")
+    if delimiter != "$":
+        raise ValueError(
+            "Index value %r does not match expected format." % (value,))
+    return (key, from_reverse_timestamp(timestamp), address)
```

Comment:
This looks identitical to `key_with_ts_and_value_formatter`, apart from the `from_reverse_timestamp()` call at the bottom. Could we not have this function just use `key_with_ts_and_value_formatter`, then then use `from_reverse_timestamp()` on the timestamp in the returned tuple?